server/reports/attendance/attendance-endpoint.js

```diff
@@ -73,64 +67,62 @@ export function attendanceRequestHandler(request, response) {
+        data: [],
+        headers: ["class", "capacity", "registered", "attended"],
+      };
       let allNumberAttendedPromises = [];
       let numberOfClasses = Object.keys(classes.Classes).length;
 
       // iterates through every class in classes
       for (let i = 0; i < numberOfClasses; ++i) {
-        let classTitle = classes.Classes[i].ClassDescription.Name;
         let classID = classes.Classes[i].Id;
         let maxCapacity = classes.Classes[i].MaxCapacity;
         let numberRegistered = classes.Classes[i].TotalBooked;
-        //let classDate = fixDateFormat(classes.Classes[i].StartDateTime); // Mindbody StartDateTime format is yyyy/mm/dd by default
+        let classDate = fixDateFormat(classes.Classes[i].StartDateTime); // Mindbody StartDateTime format is yyyy/mm/dd by default
+
+        // adds class data to attendance report if startdate <= classdate <= enddate
+        if (isValidDate(startdate, classDate, enddate)) {
+          let classData = {
+            class: classID,
+            capacity: maxCapacity,
+            registered: numberRegistered,
+            attended: 0
+          };
+
+          // adds attendance parameter to classData
+          let numberAttendedPromise = getNumberAttended(classID)
+            .then((numberAttended) => {
+              classData["attended"] = numberAttended;
+            })
+            .catch((err) => {
+              console.log("ERROR! ", err);
+            });
+          allNumberAttendedPromises.push(numberAttendedPromise);
+          attendanceReport.data.push([classData.class, classData.capacity, classData.registered, classData.attended]); // pushes current class's data to attendanceReport
+        }
+      }
 
-        let classData = {
-          class_title: classTitle,
-          class_ID: classID,
-          capacity: maxCapacity,
-          registered: numberRegistered,
-        };
-        // adds attendance parameter to classData
-        let numberAttendedPromise = getNumberAttended(classID)
-          .then((numberAttended) => {
-            classData["attended"] = numberAttended;
-          })
-          .catch((err) => {
-            // temporarily commented out since this always executes, and for every iteraetion of the loop
-            // figure out how to avoid this since the promise does resolve later and the data is caught and output
-            //console.log("ERROR! ", err);
-          });
-        allNumberAttendedPromises.push(numberAttendedPromise);
-        attendanceReport.push(classData); // pushes current class's data to attendanceReport
-      } 
       // resolves all, allNumberAttendedPromises then outputs attendance report to endpoint as CSV or JSON
       Promise.all(allNumberAttendedPromises)
         .then(() => {
           if (format === "csv") {
-            let fields = ["class_title", "class_ID", "capacity", "registered", "attended"];
+            let fields = ["class", "capacity", "registered", "attended"];
```

Comment:
Any specific reason these got combined? I know we talked about changing some of the headers in meeting, but I think class_id is probably a good to have based on what mindbody is reporting through their existing reports. They seem to be dumping at least clientIDs to the frontend, so I feel it is somewhat likely that Shannon actually has that info somewhere, and may wish to perform additional steps on it later.